addons/mail/models/mail_alias.py

```diff
@@ -134,12 +134,15 @@ def _find_unique(self, name, alias_ids=False):
            an unused alias is found.
         """
         sequence = None
+        icp = self.env['ir.config_parameter'].sudo()
+        catchall_alias = icp.get_param('mail.catchall.alias')
+        bounce_alias = icp.get_param('mail.bounce.alias')
         while True:
             new_name = "%s%s" % (name, sequence) if sequence is not None else name
             domain = [('alias_name', '=', new_name)]
             if alias_ids:
                 domain += [('id', 'not in', alias_ids)]
-            if not self.search(domain):
+            if not self.search(domain) and new_name not in [catchall_alias, bounce_alias]:
```

Comment:
You could inverse the two condition, so that we don't perform the search if we know we have a conflict.